### PR TITLE
[11.0][account_banking_mandate] fixes to support multicompany

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
The following fixes are required when you operate in a multicompany environment.

### CODE REVIEW

File: account_banking_mandate/models/account_banking_mandate.py [hunk 1]
```diff
@@ -157,11 +157,25 @@ def create(self, vals=None):
...
         return super(AccountBankingMandate, self).create(vals)
 
-    @api.multi
+    @api.onchange('company_id')
+    def _onchange_company_id(self):
+        res = {}
+        partner_domain = self._get_default_partner_bank_id_domain()
+        if self.company_id:
+            if self.partner_bank_id and \
+                    self.partner_bank_id.company_id != self.company_id:
```
Comment (etobella) [inline account_banking_mandate/models/account_banking_mandate.py:168]: and self.company_id and self.partner_bank_id.company_id

**Review by etobella — CHANGES_REQUESTED** [review]

File: account_banking_mandate/models/account_banking_mandate.py [hunk 2]
```diff
@@ -157,11 +157,27 @@ def create(self, vals=None):
...
+            domain = {
+                'partner_bank_id': partner_domain}
+        res['domain'] = domain
+        return res
+
     @api.onchange('partner_bank_id')
     def mandate_partner_bank_change(self):
-        for mandate in self:
-            mandate.partner_id = mandate.partner_bank_id.partner_id
+        self.partner_id = self.partner_bank_id.partner_id
```
Comment (MiquelRForgeFlow) [inline account_banking_mandate/models/account_banking_mandate.py:180]: not sure, but shouldn't you add `if self.partner_bank_id.partner_id:` ?

Comment (JordiBForgeFlow) [reply account_banking_mandate/models/account_banking_mandate.py:180]: Well, if there's no partner then self.partner_id = False. No need for the if

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Functional test on runbot

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

**Pull request merged**